src/Termonad/Types.hs

```diff
@@ -530,6 +532,23 @@ data TMStateInvariantErr
   | TabsDoNotMatch TabsDoNotMatch
   deriving Show
 
+printWidgetTree :: Gtk.IsWidget a => a -> IO ()
+printWidgetTree widget_ = do
+  widget <- Gtk.toWidget widget_
+  go "" widget
+  where
+    go :: Text -> Gtk.Widget -> IO ()
+    go indent w = do
+      type_ <- gtypeFromInstance w
+      name <- Gtk.gtypeName type_
+      let ptr = Gtk.managedForeignPtr . Gtk.toManagedPtr $ w
+      putStrLn $ indent <> pack name <> "  " <> pack (show ptr)
+      maybeContainer <- Gtk.castTo Gtk.Container w
+      for_ maybeContainer $ \container -> do
+        children <- Gtk.containerGetChildren container
+        for_ children $ \child -> do
+          go ("  " <> indent) child
```

Comment:
This is really nice.  I've wanted something like this before, so I opened an issue on haskell-gi to see if it could potentially be included in `gi-gtk` or somewhere.

I don't think it makes sense to only have this in the Termonad codebase, since it seems widely helpful.

Although, if this type of function is not accepted upstream, let's move it to the `Termonad.Gtk` module, since there are already a few other helper functions in there.

Reply:
I have now moved `printWidgetTree` to `Termonad.Gtk`, let's remove when/if it is accepted upstream.